File: chrome/test/data/webui/gaia_auth_host/gaia_auth_host_browsertest.js
// Copyright 2022 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

/**
 * @fileoverview Test fixture for c/b/r/gaia_auth_host.
 */

GEN('#include "content/public/test/browser_test.h"');

var GaiaAuthHostBrowserTest = class extends testing.Test {
  /** @override */
  get browsePreload() {
    return 'chrome://dummyurl';
  }

  /** @override */
  get isAsync() {
    return true;
  }
};

[['PasswordChangeAuthenticator', 'password_change_authenticator_test.js'],
 ['SamlPasswordAttributes', 'saml_password_attributes_test.js'],
 ['SamlTimestamps', 'saml_timestamps_test.js'],
 ['SamlUsernameAutofill', 'saml_username_autofill_test.js'],
].forEach(test => registerTest(...test));

// Common WebUI pattern for test registration.
function registerTest(testName, module, caseName) {
  const className = `GaiaAuthHost${testName}Test`;
  this[className] = class extends GaiaAuthHostBrowserTest {
    /** @override */
    get browsePreload() {
      return `chrome://chrome-signin/test_loader.html?module=gaia_auth_host/${
          module}`;
    }
  };

  TEST_F(className, caseName || 'All', () => mocha.run());
}

File: chrome/test/data/webui/gaia_auth_host/gaia_auth_host_browsertest.cc
// Copyright 2023 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "base/strings/stringprintf.h"
#include "chrome/common/webui_url_constants.h"
#include "chrome/test/base/web_ui_mocha_browser_test.h"
#include "components/signin/public/base/signin_metrics.h"
#include "content/public/test/browser_test.h"

class GaiaAuthHostTest : public WebUIMochaBrowserTest {
 protected:
  GaiaAuthHostTest() {
    set_test_loader_host(chrome::kChromeUIChromeSigninHost);
  }

  void RunTestCase(const std::string& file_name) {
    WebUIMochaBrowserTest::RunTest(
        base::StringPrintf(
            "%s&reason=%d", file_name.c_str(),
            static_cast<int>(signin_metrics::Reason::kFetchLstOnly)),
        "mocha.run()");
  }
};

IN_PROC_BROWSER_TEST_F(GaiaAuthHostTest, PasswordChangeAuthenticator) {
  RunTestCase("gaia_auth_host/password_change_authenticator_test.js");
}

IN_PROC_BROWSER_TEST_F(GaiaAuthHostTest, SamlPasswordAttributes) {
  RunTestCase("gaia_auth_host/saml_password_attributes_test.js");
}

IN_PROC_BROWSER_TEST_F(GaiaAuthHostTest, SamlTimestamps) {
  RunTestCase("gaia_auth_host/saml_timestamps_test.js");
}

IN_PROC_BROWSER_TEST_F(GaiaAuthHostTest, SamlUsernameAutofill) {
  RunTestCase("gaia_auth_host/saml_username_autofill_test.js");
}
